src/pages/index.js

```diff
@@ -107,18 +131,80 @@ profileAddButton.addEventListener("click", function () {
 
 /* Functions */
 
-// Define the createCard function
-function createCard(cardData) {
-  // Select the template element and clone its contents
-  const cardInfo = new Card(cardData, "#card__template", function () {
-    openZoomPicture(cardData.link, cardData.name);
+function handleAvatarUpdate({ link }) {
+  return api.updateAvatar({ avatar: link }).then((res) => {
+    userInfo.setUserInfo({
+      title: res.name,
+      description: res.about,
+      avatar: res.avatar,
+    });
   });
+}
+
+function cardLikeClick(cardId, isLiked) {
+  if (isLiked) {
+    return api.likeCard(cardId).catch((error) => {
+      console.error(`Failed to like card with ID ${cardId}:`, error);
```

Comment:
I you have `catch` inside `Card`, then you don't need it here, otherwise the code will proceed after this `catch`